petastorm/tests/test_tf_dataset.py

```diff
@@ -121,6 +121,34 @@ def test_with_dataset_repeat(synthetic_dataset, reader_factory):
                 sess.run(iterator)
 
 
+@pytest.mark.forked
+@pytest.mark.parametrize('reader_factory', ALL_READER_FLAVOR_FACTORIES)
+@create_tf_graph
+def test_with_dataset_repeat_after_cache(synthetic_dataset, reader_factory):
```

Comment:
Not for this PR, but we should probably try using modern TF interface (here `import tensorflow.compat.v1 as tf`). That way the tests would become less bloated as we would work with eager execution mode.

Reply:
I agree, it is very painful to debug without eager execution mode.